components/brave_rewards/browser/database/publisher_info_database.cc

```diff
@@ -1607,11 +1702,85 @@ bool PublisherInfoDatabase::Migrate(int version) {
     case 7: {
       return MigrateV6toV7();
     }
+    case 8: {
+      return MigrateV7toV8();
+    }
     default:
       return false;
   }
 }
 
+bool PublisherInfoDatabase::MigrateDBTable(
+    const std::string& from,
+    const std::string& to,
+    const std::vector<std::string>& columns,
+    const bool should_drop) {
+  std::map<std::string, std::string> new_columns;
+  for (const auto& column : columns) {
+    new_columns[column] = column;
+  }
+
+  return MigrateDBTable(from, to, new_columns, should_drop);
+}
+
+bool PublisherInfoDatabase::MigrateDBTable(
+    const std::string& from,
+    const std::string& to,
+    const std::map<std::string, std::string>& columns,
+    const bool should_drop) {
+  DCHECK_NE(from, to);
+  DCHECK(!from.empty());
+  DCHECK(!to.empty());
+
+  std::string sql = "PRAGMA foreign_keys = off;";
+
+  if (!columns.empty()) {
+    const auto insert = CreateDBInsertQuery(from, to, columns);
+    sql.append(insert);
+  }
+
+  if (should_drop) {
+    sql.append(base::StringPrintf("DROP TABLE %s;", from.c_str()));
+  }
+
+  sql.append("PRAGMA foreign_keys = on;");
+
+  return GetDB().Execute(sql.c_str());
+}
+
+bool PublisherInfoDatabase::RenameDBTable(
+    const std::string& from,
+    const std::string& to) {
+  DCHECK_NE(from, to);
+
+  const auto sql = base::StringPrintf("ALTER TABLE %s RENAME TO %s;",
+      from.c_str(), to.c_str());
+
+  return GetDB().Execute(sql.c_str());
+}
+
+std::string PublisherInfoDatabase::CreateDBInsertQuery(
```

Comment:
Not sure if we need `CreateDB` at the beginning as you are not creating db in this function. Maybe `SelectAndInsertQuery`?

Reply:
Renamed to `GenerateDBInsertQuery`